spec/system/search_spec.rb

```diff
@@ -36,4 +36,45 @@
       expect(search_page.heading_text).to eq("Search")
     end
   end
+
+  describe "when using full page search on desktop" do
+    before do
+      SearchIndexer.enable
+      SearchIndexer.index(topic, force: true)
+      RateLimiter.enable
+    end
+
+    after { SearchIndexer.disable }
+
+    it "rate limits searches for anonymous users" do
+      queries = %w[
+        one
+        two
+        three
+        four
+        five
+        six
+        seven
+        eight
+        nine
+        ten
+        eleven
+        twelve
+        thirteen
+        fourteen
+        fifteen
+      ]
+
+      visit("/search?expanded=true")
+
+      queries.each do |query|
+        search_page.clear_search_input
+        search_page.type_in_search(query)
+        search_page.click_search_button
+      end
+
+      # Rate limit error should kick in after 15 queries
```

Comment:
I guess this relies on `rate_limit_search_anon_user_per_minute` being set to 15 by default. It would be good to be more explicit here and set `SiteSetting.rate_limit_search_anon_user_per_minute` in the before block and then check against that.